go/stellar/stellarsvc/frontend.go

```diff
@@ -239,16 +244,218 @@ func (s *Server) LinkNewWalletAccountLocal(ctx context.Context, arg stellar1.Lin
+	ctx = s.logTag(ctx)
+	defer s.G().CTraceTimed(ctx, "GetPaymentsLocal", func() error { return err })()
+	err = s.assertLoggedIn(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	srvPayments, err := s.remoter.RecentPayments(ctx, arg.AccountID, 0)
+	if err != nil {
+		return nil, err
+	}
+	payments = make([]stellar1.PaymentOrErrorLocal, len(srvPayments))
+	for i, p := range srvPayments {
+		payments[i].Payment, err = s.transformPaymentSummary(ctx, arg.AccountID, p)
+		if err != nil {
+			s := err.Error()
+			payments[i].Err = &s
+			payments[i].Payment = nil // just to make sure
+		}
+	}
+
+	return payments, nil
+}
+
+func (s *Server) transformPaymentSummary(ctx context.Context, acctID stellar1.AccountID, p stellar1.PaymentSummary) (*stellar1.PaymentLocal, error) {
+	typ, err := p.Typ()
+	if err != nil {
+		return nil, err
+	}
+
+	switch typ {
+	case stellar1.PaymentSummaryType_STELLAR:
+		return s.transformPaymentStellar(ctx, acctID, p.Stellar())
+	case stellar1.PaymentSummaryType_DIRECT:
+		return s.transformPaymentDirect(ctx, acctID, p.Direct())
+	case stellar1.PaymentSummaryType_RELAY:
+		return s.transformPaymentRelay(ctx, acctID, p.Relay())
+	default:
+		return nil, fmt.Errorf("unrecognized payment type: %s", typ)
+	}
+}
+
+func (s *Server) transformPaymentStellar(ctx context.Context, acctID stellar1.AccountID, p stellar1.PaymentSummaryStellar) (*stellar1.PaymentLocal, error) {
+	loc, err := newPaymentLocal(p.TxID, p.Ctime, p.Amount, p.From, p.To, acctID)
+	if err != nil {
+		return nil, err
+	}
+
+	loc.Source = p.From.String()
+	loc.SourceType = ParticipantTypeStellar
+	loc.Target = p.To.String()
+	loc.TargetType = ParticipantTypeStellar
+
+	loc.Status = stellar1.PaymentStatus_COMPLETED
+	loc.StatusDescription = strings.ToLower(loc.Status.String())
+
+	return loc, nil
+}
+
+func (s *Server) transformPaymentDirect(ctx context.Context, acctID stellar1.AccountID, p stellar1.PaymentSummaryDirect) (*stellar1.PaymentLocal, error) {
+	loc, err := newPaymentLocal(p.TxID, p.Ctime, p.Amount, p.FromStellar, p.ToStellar, acctID)
+	if err != nil {
+		return nil, err
+	}
+
+	loc.Worth, loc.WorthCurrency, err = s.formatWorth(ctx, p.DisplayAmount, p.DisplayCurrency)
+	if err != nil {
+		return nil, err
+	}
+
+	loc.Source, loc.SourceType = s.lookupUsernameFallback(ctx, p.From.Uid, p.FromStellar)
+
+	if p.To != nil {
+		loc.Target, loc.TargetType = s.lookupUsernameFallback(ctx, p.To.Uid, p.ToStellar)
+	}
+
+	loc.Status = p.TxStatus.ToPaymentStatus()
+	loc.StatusDescription = strings.ToLower(loc.Status.String())
+	loc.StatusDetail = p.TxErrMsg
+
+	loc.Note, loc.NoteErr = s.decryptNote(ctx, p.TxID, p.NoteB64)
+
+	return loc, nil
+}
+
+func (s *Server) transformPaymentRelay(ctx context.Context, acctID stellar1.AccountID, p stellar1.PaymentSummaryRelay) (*stellar1.PaymentLocal, error) {
+	loc, err := newPaymentLocal(p.TxID, p.Ctime, p.Amount, p.FromStellar, "", acctID)
```

Comment:
If you successfully claimed the relay you should get a `_INCREASE`. As it is I think you'll get `_NONE`.